app/src/main/java/com/example/instagram/LoginActivity.java

```diff
@@ -0,0 +1,66 @@
+package com.example.instagram;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.parse.LogInCallback;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+
+public class LoginActivity extends AppCompatActivity {
+
+    public static final String TAG = "LoginActivity";
+    private EditText etUsername;
+    private EditText etPassword;
+    private Button btnLogin;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        etUsername = findViewById(R.id.etUsername);
+        etPassword = findViewById(R.id.etPassword);
+        btnLogin = findViewById(R.id.btnLogin);
+        btnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.i(TAG, "onClick login button");
+                String username = etUsername.getText().toString();
+                String password = etPassword.getText().toString();
+                loginUser(username, password);
+
+            }
+        });
+
+    }
+
+    private void loginUser(String username, String password) {
+        Log.i(TAG, "Attempting to login user " + username);
+        ParseUser.logInInBackground(username, password, new LogInCallback() {
+            @Override
+            public void done(ParseUser user, ParseException e) {
+                if (e != null) {
+                    Log.e(TAG, "Issue with login", e);
+                    return;
+                }
+                goMainActivity();
```

Comment:
Perhaps `navigateToMainActivity()` is a better name.

Btw, in case you were wondering about the slightly weird code path here:
1. `goMainActivity()` starts a new Activity and calls `finish()` on this one. This means that this current Activity will be destroyed.
2. There's a Log and Toast statement (Line 55 and 56) that run after the `finish()` is called. You should not not see the Log and Toast because the Activity is destroyed by this point. But I'm sure you see them, why does this happen?

This is because the `finish()` method sends a request to the Android framework to destroy the Activity. The framework will add that request to its queue and get to it eventually but not immediately. This gives time for the Log and Toast statements to run.

This is absolutely fine in this case, but you don't want to rely on this delay too much. If you kick off some heavy process after `finish()` is called, it is not guaranteed to run to completion.